impl/src/main/java/org/jboss/weld/bootstrap/BeanDeployment.java

```diff
@@ -271,6 +273,11 @@ public void createBeans(Environment environment) {
             beanDeployer.addBuiltInBean(ContextBean.of(context, beanManager));
         }
 
+        if (beanDeploymentArchive.getBeansXml() instanceof TrimmableBeansXml) {
+            if (((TrimmableBeansXml) beanDeploymentArchive.getBeansXml()).isTrimmed()) {
```

Comment:
This is not very readable... I would use something like:
```
   TrimmableBeansXml trimmableBeansXml  = ...
   if (trimmableBeansXml.isTrimmed()) {
     ...
   }
```
But that's just minor.... otherwise the PR looks good!